### PR TITLE
[Doc] [Serve] Add docs for custom resources with Serve

### LINKED ISSUE #28477: [Serve] [Docs] Document `ray_actor_options` options
### Description

`ray_actor_options` allows users to configure [an allow-listed subset](https://github.com/ray-project/ray/blob/4daefb28f5ccfdd14681cf8590e098f9a6744d67/python/ray/serve/config.py#L394-L404) of general actor options. However, this isn't documented anywhere.

The documentation explicitly list which options are supported by `ray_actor_options`. It should also enumerate which `accelerator_types` are compatible with Ray Serve. Users on the Discuss forum ([example](https://discuss.ray.io/t/how-can-i-specify-accelerator-type-for-ray-serve/7531)) have asked about this.

### Link

N/A

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
- Adds an example to the docs about using custom resources with Ray Serve
- Documents which options are allowed for `ray_actor_options`

## Related issue number

Closes #28477 
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -317,7 +317,9 @@ def deployment(
             a '/' unless they're the root (just '/'), which acts as a
             catch-all.
         ray_actor_options: Options to be passed to the Ray actor
-            constructor such as resource requirements.
+            constructor such as resource requirements. Valid options are
+            `accellerator_type`, `memory`, `num_cpus`, `num_gpus`,
```
Comment (shrekris-anyscale) [inline python/ray/serve/api.py:321]:
```suggestion
            `accelerator_type`, `memory`, `num_cpus`, `num_gpus`,
```

**Review by shrekris-anyscale — COMMENTED** [review]
Nice work! Can we also explicitly list
  1. The `ray_actor_options` options somewhere in the docs (instead of only in the docstring)?
  2. The `accelerator` types that are allowed? I believe only the accelerators from [accelerators.py](https://github.com/ray-project/ray/blob/8be5f016afefb2e199fa45416a8c2021e05805e0/python/ray/util/accelerators/accelerators.py) are allowed. @simon-mo do you know if other accelerators can be used?

**Review by sihanwang41 — APPROVED** [review]

**Review by simon-mo — APPROVED** [review]

**Comment by architkulkarni** [comment]
@shrekris-anyscale Thanks, I've just linked to the existing accelerator doc which seems to say that those are the only available types.  Ready for review again

**Review by shrekris-anyscale — APPROVED** [review]
Nice work!

**Comment by architkulkarni** [comment]
@ray-project/ray-docs Could you please give codeowner approval?

**Review by ericl — APPROVED** [review]

**Pull request merged**